webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -855,28 +856,41 @@ class PolicyWriterServiceImpl(
       UIO.unit
     } else {
       for {
-        mvOptions  <- getMoveOptions(sortedFolder.head)
+        _          <- getMoveOptions(sortedFolder.head)
         newFolders <- Ref.make(List.empty[NodePoliciesPaths])
-                      // can't trivialy parallelise because we need parents before children
-        _          <- ZIO.foreachParN_(maxParallelism)(sortedFolder) { case folder @ NodePoliciesPaths(_, baseFolder, newFolder, backupFolder) =>
+                      // can't trivialy parallelise because we need parents before
+                      // here, base folder, newFolder and backupFolder target agent directory under rules: we need one level up appart for root.
+        _          <- (ZIO.foreachParN_(maxParallelism)(sortedFolder) { case folder @ NodePoliciesPaths(nodeId, baseFolderAgent, newFolderAgent, backupFolderAgent) =>
+                        val (baseFolder, newFolder, backupFolder) = nodeId match {
+                          case Constants.ROOT_POLICY_SERVER_ID =>
+                            // root server doesn't have the "rule" middle directory, don't get parent
+                            (File(baseFolderAgent), File(newFolderAgent), backupFolderAgent.map(x => File(x)))
+                          case _                               =>
+                            // get parent of /var/rudder/share/xxxx-xxxx-xxxx/rules/cfengine-community
+                            (File(baseFolderAgent).parent, File(newFolderAgent).parent, backupFolderAgent.map(x => File(x).parent))
+                        }
                         for {
-                          _ <- PolicyGenerationLoggerPure.trace(s"Backuping old policies from '${baseFolder}' to '${backupFolder} ")
-                          _ <- backupNodeFolder(baseFolder, backupFolder, mvOptions)
-                          _ <- newFolders.update( folder :: _ )
+                          _ <- PolicyGenerationLoggerPure.trace(s"Backuping old policies from '${baseFolder}' to '${backupFolder}' ")
+                          _ <- backupNodeFolder(baseFolder, backupFolder)
+                        //  _ <- newFolders.update( folder :: _ )
```

Comment:
You don't add folder anymore here

Reply:
rollback in error won't be done anymore, so either no rollback or update newFolders ref